procurement_mto_analytic/models/stock_rule.py

```diff
@@ -0,0 +1,39 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import json
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _find_matching_analytic_distribution_record(self, analytic_distribution):
+        # Prepare the JSONB structure as a string for the SQL query
+        analytic_distribution_str = json.dumps(analytic_distribution)
```

Comment:
@yostashiro I checked the behavior of how analytic_distribution sequence works. We don't need to worry about the sequence. It is always sequence by projects and multiple analytic_distributions for one project are also sequence by themselves even we don't fill in order.